### ISSUE TITLE
use-of-uninitialized-value in llvm::MachineBasicBlock::~MachineBasicBlock()

### ISSUE THREAD

**Issue by Avogar** [ClickHouse/ClickHouse#42505]
https://s3.amazonaws.com/clickhouse-test-reports/42130/20dbf5eb3404b7473d70c6e6a48fc844d3d8b14c/stress_test__msan_/stderr.log

```
==1441==WARNING: MemorySanitizer: use-of-uninitialized-value
    #0 0x4a6a9fd9 in llvm::iplist_impl<llvm::simple_ilist<llvm::MachineInstr, llvm::ilist_sentinel_tracking<true>>, llvm::ilist_traits<llvm::MachineInstr>>::~iplist_impl() MachineBasicBlock.cpp
    #1 0x4a6aa3ff in llvm::MachineBasicBlock::~MachineBasicBlock() (/usr/bin/clickhouse+0x4a6aa3ff) (BuildId: df7b79e74b9045bcc07be9a7f5b28c9d68ebe310)
    #2 0x4a822fbf in llvm::ilist_alloc_traits<llvm::MachineBasicBlock>::deleteNode(llvm::MachineBasicBlock*) (/usr/bin/clickhouse+0x4a822fbf) (BuildId: df7b79e74b9045bcc07be9a7f5b28c9d68ebe310)
    #3 0x4a6d6036 in llvm::MachineBasicBlock::eraseFromParent() (/usr/bin/clickhouse+0x4a6d6036) (BuildId: df7b79e74b9045bcc07be9a7f5b28c9d68ebe310)
    #4 0x4afe7147 in llvm::TailDuplicator::removeDeadBlock(llvm::MachineBasicBlock*, llvm::function_ref<void (llvm::MachineBasicBlock*)>*) (/usr/bin/clickhouse+0x4afe7147) (BuildId: df7b79e74b9045bcc07be9a7f5b28c9d68ebe310)
    #5 0x4afdef41 in llvm::TailDuplicator::tailDuplicateAndUpdate(bool, llvm::MachineBasicBlock*, llvm::MachineBasicBlock*, llvm::SmallVectorImpl<llvm::MachineBasicBlock*>*, llvm::function_ref<void (llvm::MachineBasicBlock*)>*, llvm::SmallVectorImpl<llvm::MachineBasicBlock*>*) (/usr/bin/clickhouse+0x4afdef41) (BuildId: df7b79e74b9045bcc07be9a7f5b28c9d68ebe310)
    #6 0x4afe7959 in llvm::TailDuplicator::tailDuplicateBlocks() (/usr/bin/clickhouse+0x4afe7959) (BuildId: df7b79e74b9045bcc07be9a7f5b28c9d68ebe310)
    #7 0x4afdb3bc in (anonymous namespace)::TailDuplicateBase::runOnMachineFunction(llvm::MachineFunction&) TailDuplication.cpp
    #8 0x4a860be3 in llvm::MachineFunctionPass::runOnFunction(llvm::Function&) (/usr/bin/clickhouse+0x4a860be3) (BuildId: df7b79e74b9045bcc07be9a7f5b28c9d68ebe310)
    #9 0x4f7302ad in llvm::FPPassManager::runOnFunction(llvm::Function&) (/usr/bin/clickhouse+0x4f7302ad) (BuildId: df7b79e74b9045bcc07be9a7f5b28c9d68ebe310)
    #10 0x4f759901 in llvm::FPPassManager::runOnModule(llvm::Module&) (/usr/bin/clickhouse+0x4f759901) (BuildId: df7b79e74b9045bcc07be9a7f5b28c9d68ebe310)
    #11 0x4f734b52 in llvm::legacy::PassManagerImpl::run(llvm::Module&) (/usr/bin/clickhouse+0x4f734b52) (BuildId: df7b79e74b9045bcc07be9a7f5b28c9d68ebe310)
    #12 0x4f75a6eb in llvm::legacy::PassManager::run(llvm::Module&) (/usr/bin/clickhouse+0x4f75a6eb) (BuildId: df7b79e74b9045bcc07be9a7f5b28c9d68ebe310)
    #13 0x41f70eff in DB::JITCompiler::compile(llvm::Module&) build_docker/../src/Interpreters/JIT/CHJIT.cpp:78:22
    #14 0x41f6c47d in DB::CHJIT::compileModule(std::__1::unique_ptr<llvm::Module, std::__1::default_delete<llvm::Module>>) build_docker/../src/Interpreters/JIT/CHJIT.cpp:378:29
    #15 0x41f6b6ed in DB::CHJIT::compileModule(std::__1::function<void (llvm::Module&)>) build_docker/../src/Interpreters/JIT/CHJIT.cpp:359:24
    #16 0x41fab2e3 in DB::compileAggregateFunctions(DB::CHJIT&, std::__1::vector<DB::AggregateFunctionWithOffset, std::__1::allocator<DB::AggregateFunctionWithOffset>> const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>) build_docker/../src/Interpreters/JIT/compileFunction.cpp:769:32
    #17 0x3e6b1f1a in DB::Aggregator::compileAggregateFunctionsIfNeeded()::$_0::operator()() const build_docker/../src/Interpreters/Aggregator.cpp:693:49
    #18 0x3e6b1f1a in std::__1::pair<std::__1::shared_ptr<DB::CompiledExpressionCacheEntry>, bool> DB::CacheBase<wide::integer<128ul, unsigned int>, DB::CompiledExpressionCacheEntry, UInt128Hash, DB::CompiledFunctionWeightFunction>::getOrSet<DB::Aggregator::compileAggregateFunctionsIfNeeded()::$_0>(wide::integer<128ul, unsigned int> const&, DB::Aggregator::compileAggregateFunctionsIfNeeded()::$_0&&) build_docker/../src/Common/CacheBase.h:125:24
    #19 0x3e6b1f1a in DB::Aggregator::compileAggregateFunctionsIfNeeded() build_docker/../src/Interpreters/Aggregator.cpp:689:70
    #20 0x3e6a9cf5 in DB::Aggregator::Aggregator(DB::Block const&, DB::Aggregator::Params const&) build_docker/../src/Interpreters/Aggregator.cpp:626:5
    #21 0x45f9f9b3 in DB::Aggregator* std::__1::construct_at<DB::Aggregator, DB::Block const&, DB::Aggregator::Params&, DB::Aggregator*>(DB::Aggregator*, DB::Block const&, DB::Aggregator::Params&) build_docker/../contrib/libcxx/include/__memory/construct_at.h:38:50
    #22 0x45f9f9b3 in void std::__1::allocator_traits<std::__1::allocator<std::__1::__list_node<DB::Aggregator, void*>>>::construct<DB::Aggregator, DB::Block const&, DB::Aggregator::Params&, void, void>(std::__1::allocator<std::__1::__list_node<DB::Aggregator, void*>>&, DB::Aggregator*, DB::Block const&, DB::Aggregator::Params&) build_docker/../contrib/libcxx/include/__memory/allocator_traits.h:298:9
    #23 0x45f9f9b3 in std::__1::__list_iterator<DB::Aggregator, void*> std::__1::list<DB::Aggregator, std::__1::allocator<DB::Aggregator>>::emplace<DB::Block const&, DB::Aggregator::Params&>(std::__1::__list_const_iterator<DB::Aggregator, void*>, DB::Block const&, DB::Aggregator::Params&) build_docker/../contrib/libcxx/include/list:1648:5
    #24 0x45f9f9b3 in DB::AggregatingTransformParams::AggregatingTransformParams(DB::Block const&, DB::Aggregator::Params const&, bool) build_docker/../src/Processors/Transforms/AggregatingTransform.h:41:44
    #25 0x45f9d660 in DB::AggregatingTransformParams* std::__1::construct_at<DB::AggregatingTransformParams, DB::Block const&, DB::Aggregator::Params, bool&, DB::AggregatingTransformParams*>(DB::AggregatingTransformParams*, DB::Block const&, DB::Aggregator::Params&&, bool&) build_docker/../contrib/libcxx/include/__memory/construct_at.h:38:50
    #26 0x45f9d660 in void std::__1::allocator_traits<std::__1::allocator<DB::AggregatingTransformParams>>::construct<DB::AggregatingTransformParams, DB::Block const&, DB::Aggregator::Params, bool&, void, void>(std::__1::allocator<DB::AggregatingTransformParams>&, DB::AggregatingTransformParams*, DB::Block const&, DB::Aggregator::Params&&, bool&) build_docker/../contrib/libcxx/include/__memory/allocator_traits.h:298:9
    #27 0x45f9d660 in std::__1::__shared_ptr_emplace<DB::AggregatingTransformParams, std::__1::allocator<DB::AggregatingTransformParams>>::__shared_ptr_emplace<DB::Block const&, DB::Aggregator::Params, bool&>(std::__1::allocator<DB::AggregatingTransformParams>, DB::Block const&, DB::Aggregator::Params&&, bool&) build_docker/../contrib/libcxx/include/__memory/shared_ptr.h:293:9
    #28 0x45f9d660 in std::__1::shared_ptr<DB::AggregatingTransformParams> std::__1::allocate_shared<DB::AggregatingTransformParams, std::__1::allocator<DB::AggregatingTransformParams>, DB::Block const&, DB::Aggregator::Params, bool&, void>(std::__1::allocator<DB::AggregatingTransformParams> const&, DB::Block const&, DB::Aggregator::Params&&, bool&) build_docker/../contrib/libcxx/include/__memory/shared_ptr.h:954:55
    #29 0x45f9d660 in std::__1::shared_ptr<DB::AggregatingTransformParams> std::__1::make_shared<DB::AggregatingTransformParams, DB::Block const&, DB::Aggregator::Params, bool&, void>(DB::Block const&, DB::Aggregator::Params&&, bool&) build_docker/../contrib/libcxx/include/__memory/shared_ptr.h:963:12
    #30 0x45f851b0 in DB::AggregatingStep::transformPipeline(DB::QueryPipelineBuilder&, DB::BuildQueryPipelineSettings const&) build_docker/../src/Processors/QueryPlan/AggregatingStep.cpp:139:29
    #31 0x4608c5c7 in DB::ITransformingStep::updatePipeline(std::__1::vector<std::__1::unique_ptr<DB::QueryPipelineBuilder, std::__1::default_delete<DB::QueryPipelineBuilder>>, std::__1::allocator<std::__1::unique_ptr<DB::QueryPipelineBuilder, std::__1::default_delete<DB::QueryPipelineBuilder>>>>, DB::BuildQueryPipelineSettings const&) build_docker/../src/Processors/QueryPlan/ITransformingStep.cpp:48:9
    #32 0x460f2251 in DB::QueryPlan::buildQueryPipeline(DB::QueryPlanOptimizationSettings const&, DB::BuildQueryPipelineSettings const&) build_docker/../src/Processors/QueryPlan/QueryPlan.cpp:187:47
    #33 0x41105822 in DB::InterpreterSelectWithUnionQuery::execute() build_docker/../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:370:31
    #34 0x41d30161 in DB::executeQueryImpl(char const*, char const*, std::__1::shared_ptr<DB::Context>, bool, DB::QueryProcessingStage::Enum, DB::ReadBuffer*) build_docker/../src/Interpreters/executeQuery.cpp:697:36
    #35 0x41d4394c in DB::executeQuery(DB::ReadBuffer&, DB::WriteBuffer&, bool, std::__1::shared_ptr<DB::Context>, std::__1::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)>, std::__1::optional<DB::FormatSettings> const&) build_docker/../src/Interpreters/executeQuery.cpp:1162:30
    #36 0x44fca390 in DB::HTTPHandler::processQuery(DB::HTTPServerRequest&, DB::HTMLForm&, DB::HTTPServerResponse&, DB::HTTPHandler::Output&, std::__1::optional<DB::CurrentThread::QueryScope>&) build_docker/../src/Server/HTTPHandler.cpp:798:5
    #37 0x44fd8554 in DB::HTTPHandler::handleRequest(DB::HTTPServerRequest&, DB::HTTPServerResponse&) build_docker/../src/Server/HTTPHandler.cpp:974:9
    #38 0x4515c6c2 in DB::HTTPServerConnection::run() build_docker/../src/Server/HTTP/HTTPServerConnection.cpp:65:34
    #39 0x50cdd85d in Poco::Net::TCPServerConnection::start() build_docker/../contrib/poco/Net/src/TCPServerConnection.cpp:43:3
    #40 0x50cdebee in Poco::Net::TCPServerDispatcher::run() build_docker/../contrib/poco/Net/src/TCPServerDispatcher.cpp:115:20
    #41 0x513fa4eb in Poco::PooledThread::run() build_docker/../contrib/poco/Foundation/src/ThreadPool.cpp:199:14
    #42 0x513f5f41 in Poco::(anonymous namespace)::RunnableHolder::run() build_docker/../contrib/poco/Foundation/src/Thread.cpp:55:11
    #43 0x513f1e28 in Poco::ThreadImpl::runnableEntry(void*) build_docker/../contrib/poco/Foundation/src/Thread_POSIX.cpp:345:27
    #44 0x7f517a56c608 in start_thread (/lib/x86_64-linux-gnu/libpthread.so.0+0x8608) (BuildId: 7b4536f41cdaa5888408e82d0836e33dcf436466)
    #45 0x7f517a491132 in __clone (/lib/x86_64-linux-gnu/libc.so.6+0x11f132) (BuildId: 1878e6b475720c7c51969e69ab2d276fae6d1dee)

  Memory was marked as uninitialized
    #0 0xbc607ad in __sanitizer_dtor_callback (/usr/bin/clickhouse+0xbc607ad) (BuildId: df7b79e74b9045bcc07be9a7f5b28c9d68ebe310)
    #1 0x4a6aa3da in llvm::MachineBasicBlock::~MachineBasicBlock() (/usr/bin/clickhouse+0x4a6aa3da) (BuildId: df7b79e74b9045bcc07be9a7f5b28c9d68ebe310)
    #2 0x4a822fbf in llvm::ilist_alloc_traits<llvm::MachineBasicBlock>::deleteNode(llvm::MachineBasicBlock*) (/usr/bin/clickhouse+0x4a822fbf) (BuildId: df7b79e74b9045bcc07be9a7f5b28c9d68ebe310)
    #3 0x4a6d6036 in llvm::MachineBasicBlock::eraseFromParent() (/usr/bin/clickhouse+0x4a6d6036) (BuildId: df7b79e74b9045bcc07be9a7f5b28c9d68ebe310)
    #4 0x4afe7147 in llvm::TailDuplicator::removeDeadBlock(llvm::MachineBasicBlock*, llvm::function_ref<void (llvm::MachineBasicBlock*)>*) (/usr/bin/clickhouse+0x4afe7147) (BuildId: df7b79e74b9045bcc07be9a7f5b28c9d68ebe310)
    #5 0x4afdef41 in llvm::TailDuplicator::tailDuplicateAndUpdate(bool, llvm::MachineBasicBlock*, llvm::MachineBasicBlock*, llvm::SmallVectorImpl<llvm::MachineBasicBlock*>*, llvm::function_ref<void (llvm::MachineBasicBlock*)>*, llvm::SmallVectorImpl<llvm::MachineBasicBlock*>*) (/usr/bin/clickhouse+0x4afdef41) (BuildId: df7b79e74b9045bcc07be9a7f5b28c9d68ebe310)
    #6 0x4afe7959 in llvm::TailDuplicator::tailDuplicateBlocks() (/usr/bin/clickhouse+0x4afe7959) (BuildId: df7b79e74b9045bcc07be9a7f5b28c9d68ebe310)
    #7 0x4afdb3bc in (anonymous namespace)::TailDuplicateBase::runOnMachineFunction(llvm::MachineFunction&) TailDuplication.cpp
    #8 0x4a860be3 in llvm::MachineFunctionPass::runOnFunction(llvm::Function&) (/usr/bin/clickhouse+0x4a860be3) (BuildId: df7b79e74b9045bcc07be9a7f5b28c9d68ebe310)
    #9 0x4f7302ad in llvm::FPPassManager::runOnFunction(llvm::Function&) (/usr/bin/clickhouse+0x4f7302ad) (BuildId: df7b79e74b9045bcc07be9a7f5b28c9d68ebe310)
    #10 0x4f759901 in llvm::FPPassManager::runOnModule(llvm::Module&) (/usr/bin/clickhouse+0x4f759901) (BuildId: df7b79e74b9045bcc07be9a7f5b28c9d68ebe310)
    #11 0x4f734b52 in llvm::legacy::PassManagerImpl::run(llvm::Module&) (/usr/bin/clickhouse+0x4f734b52) (BuildId: df7b79e74b9045bcc07be9a7f5b28c9d68ebe310)
    #12 0x4f75a6eb in llvm::legacy::PassManager::run(llvm::Module&) (/usr/bin/clickhouse+0x4f75a6eb) (BuildId: df7b79e74b9045bcc07be9a7f5b28c9d68ebe310)
    #13 0x41f70eff in DB::JITCompiler::compile(llvm::Module&) build_docker/../src/Interpreters/JIT/CHJIT.cpp:78:22
    #14 0x41f6c47d in DB::CHJIT::compileModule(std::__1::unique_ptr<llvm::Module, std::__1::default_delete<llvm::Module>>) build_docker/../src/Interpreters/JIT/CHJIT.cpp:378:29
    #15 0x41f6b6ed in DB::CHJIT::compileModule(std::__1::function<void (llvm::Module&)>) build_docker/../src/Interpreters/JIT/CHJIT.cpp:359:24
    #16 0x41fab2e3 in DB::compileAggregateFunctions(DB::CHJIT&, std::__1::vector<DB::AggregateFunctionWithOffset, std::__1::allocator<DB::AggregateFunctionWithOffset>> const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>) build_docker/../src/Interpreters/JIT/compileFunction.cpp:769:32
    #17 0x3e6b1f1a in DB::Aggregator::compileAggregateFunctionsIfNeeded()::$_0::operator()() const build_docker/../src/Interpreters/Aggregator.cpp:693:49
    #18 0x3e6b1f1a in std::__1::pair<std::__1::shared_ptr<DB::CompiledExpressionCacheEntry>, bool> DB::CacheBase<wide::integer<128ul, unsigned int>, DB::CompiledExpressionCacheEntry, UInt128Hash, DB::CompiledFunctionWeightFunction>::getOrSet<DB::Aggregator::compileAggregateFunctionsIfNeeded()::$_0>(wide::integer<128ul, unsigned int> const&, DB::Aggregator::compileAggregateFunctionsIfNeeded()::$_0&&) build_docker/../src/Common/CacheBase.h:125:24
    #19 0x3e6b1f1a in DB::Aggregator::compileAggregateFunctionsIfNeeded() build_docker/../src/Interpreters/Aggregator.cpp:689:70
    #20 0x3e6a9cf5 in DB::Aggregator::Aggregator(DB::Block const&, DB::Aggregator::Params const&) build_docker/../src/Interpreters/Aggregator.cpp:626:5
    #21 0x45f9f9b3 in DB::Aggregator* std::__1::construct_at<DB::Aggregator, DB::Block const&, DB::Aggregator::Params&, DB::Aggregator*>(DB::Aggregator*, DB::Block const&, DB::Aggregator::Params&) build_docker/../contrib/libcxx/include/__memory/construct_at.h:38:50
    #22 0x45f9f9b3 in void std::__1::allocator_traits<std::__1::allocator<std::__1::__list_node<DB::Aggregator, void*>>>::construct<DB::Aggregator, DB::Block const&, DB::Aggregator::Params&, void, void>(std::__1::allocator<std::__1::__list_node<DB::Aggregator, void*>>&, DB::Aggregator*, DB::Block const&, DB::Aggregator::Params&) build_docker/../contrib/libcxx/include/__memory/allocator_traits.h:298:9
    #23 0x45f9f9b3 in std::__1::__list_iterator<DB::Aggregator, void*> std::__1::list<DB::Aggregator, std::__1::allocator<DB::Aggregator>>::emplace<DB::Block const&, DB::Aggregator::Params&>(std::__1::__list_const_iterator<DB::Aggregator, void*>, DB::Block const&, DB::Aggregator::Params&) build_docker/../contrib/libcxx/include/list:1648:5
    #24 0x45f9f9b3 in DB::AggregatingTransformParams::AggregatingTransformParams(DB::Block const&, DB::Aggregator::Params const&, bool) build_docker/../src/Processors/Transforms/AggregatingTransform.h:41:44
```

**Comment by azat** [comment]
This is likely already fixed by #42123

**Comment by alexey-milovidov** [comment]
Let's assume that.
